### PR TITLE
Hotfix/0.24.2

### PR DESCRIPTION
**Pull request by jessemyers**
Spread out error handling

### CODE REVIEW

File: microcosm_pubsub/dispatcher.py [hunk 1]
```diff
@@ -76,34 +83,27 @@ def handle_message(self, media_type, message):
                 )
                 return handler_with_context(message)
             except SkipMessage as skipped:
-                self.logger.info(
+                logger.info(
                     "Skipping message for reason: {}".format(str(skipped)),
                     extra=skipped.extra,
                 )
                 return False
+            except Nack:
```
Comment (jessemyers) [inline microcosm_pubsub/dispatcher.py]: We should not use `instanceof` on errors.

File: microcosm_pubsub/dispatcher.py [hunk 2]
```diff
@@ -76,34 +83,27 @@ def handle_message(self, media_type, message):
                 )
                 return handler_with_context(message)
             except SkipMessage as skipped:
-                self.logger.info(
+                logger.info(
```
Comment (jessemyers) [inline microcosm_pubsub/dispatcher.py]: It's unclear if I should merge the message context with the extra data here.

**Comment by cdm-ium** [comment]
+1 nice thanks.

File: microcosm_pubsub/dispatcher.py (hunk 2 above)
Comment (cdm-ium) [inline microcosm_pubsub/dispatcher.py]: I would probably merge unless we add a base pubsub handler that automatically merges the context into extra. No way everyone raising the exception will remember to put the message context in there on their own.

**Pull request closed without merging**